frontend/src/metabase/query_builder/components/NativeQueryEditor/DataSourceSelectors/DataSourceSelectors.jsx

```diff
@@ -60,7 +60,14 @@ const DataSourceSelectors = ({
   requireWriteback = false,
 }) => {
   const database = query.database();
-  const databases = query.metadata().databasesList({ savedQuestions: false });
+
+  const databases = useMemo(() => {
+    const databases = query.metadata().databasesList({ savedQuestions: false });
```

Comment:
Nit: might be better to use a different variable name like `queriedDatabases` to avoid confusion with the variable defined on the previous line.